### ISSUE TITLE
Sync action silently fails with a TypeError & UnhandledPromiseRejectionWarning

### ISSUE THREAD

**Issue by ianjevans** [readmeio/github-readme-api-sync#23]
The action appears to succeed with a green check, but if you examine the output it is failing with a TypeError.
```
Run readmeio/github-readme-sync@2.0.0
(node:2493) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'match' of undefined
    at /home/runner/work/_actions/readmeio/github-readme-sync/2.0.0/main.js:95:28
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
(node:2493) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:2493) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

This was is the relevant section of the action config:
```yaml
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@master
      - uses: readmeio/github-readme-sync@2.0.0
        with:
          readme-oas-key: xxxx
           
          # OPTIONAL CONFIG, use if necessary
          oas-file-path: './dev-ops-api/api/swagger.yaml'
          # api-version: 'v1.0.0'
```

**Comment by rsaestrela** [comment]
I am having a similar issue. After having a look on the code, it looks that this is related with API versioning. API version can be specified in the action via `api-version: 'v1.0.0'` or in the spec itself under `info > version`. Probably by uncommenting `# api-version: 'v1.0.0'` will make it work for you @erunion. However, as you described, this should force the action to fail, and not to succeed silently.

```
headers: {
   'x-readme-version': apiVersion || schema.info.version, // apiVersion,
   'x-readme-source': 'github',
},
```

My biggest concern atm, is that apparently no other versioning values than `v1.0` or `v1.0.0` are being accepted by the API and this is causing the action to fail silently. Can this be a problem with the API or I am missing some versioning configuration? I am configuring versions in Readme and set them as `main version`. I've actually tried different combinations and nothing worked. Can someone please clarify this?

This is the error returned by the API.

```
{
  "error": "INTERNAL_ERROR",
  "message": "Unknown error (E11000 duplicate key error collection: readme.apisettings index: _id_ dup key: { : ObjectId('5f43eeecbee88b48fe88e87d') })",
  "suggestion": "Sorry this error isn't more helpful… if you're seeing this, we're stumped too! But email support (support@readme.io) and we can figure it out with you.",
  "docs": "https://docs.readme.com/developers/logs/93161a74-1c3d-411a-8da0-c2f61de1e98c",
  "help": "If you need help, email support@readme.io and include the following link to your API log: 'https://docs.readme.com/developers/logs/93161a74-1c3d-411a-8da0-c2f61de1e98c'.",
  "poem": [
    "If you're seeing this error",
    "Something went wrong internally",
    "This error is our catch-all",
    "It's one you should never see!"
  ]
}
```

**Comment by kph2u2** [comment]
I'm also blocked by this error. Any update on this issue?

**Comment by kanadgupta** [comment]
Hi @ianjevans @rsaestrela @kph2u2 — thanks for bringing these issues to our attention and apologies for the delay. [`v2.1.0`](https://github.com/readmeio/github-readme-sync/releases/tag/2.1.0) of this package should address all the issues you've mentioned. We've also added debugging logs which should make troubleshooting easier going forward ([see here](https://docs.readme.com/docs/automatically-sync-api-specification-with-github) for the docs). If you have any other troubles, please feel free to open up another issue or reach out to our support team at support@readme.io. Thanks again!

**Comment by rsaestrela** [comment]
Thanks @kanadgupta!
